signin/index.js

```diff
@@ -0,0 +1,115 @@
+
+//이메일, 비밀번호 error-message
+const emailErrorMessage = document.querySelector("#error-email");
+const passwordErrorMessage = document.querySelector("#error-password");
+
+//로그인 버튼
+const loginBtn = document.querySelector(".btn-login");
+
+//비밀번호 btn-eye
+const eyeBtn = document.querySelector(".btn-eye");
+
+/********************
+ * FUNCTION
+ ********************/
+
+//error-message 토글 함수: message가 있으면 on / 없으면 off
+function toggleErrorMessage(element, message = "") {
+  const prevInput = element === emailErrorMessage ? email : password;
+  element.textContent = message;
+
+  console.log("에러 메시지 토글");
+
+  if (message) {
+    element.classList.remove("display-none");
+    prevInput.classList.add("input-error");
+  } else {
+    element.classList.add("display-none");
+    prevInput.classList.remove("input-error");
+  }
+}
+
+//email, password 유효성 검사
+function checkForm(element) {
+  const errorMessage = element === email ? emailErrorMessage : passwordErrorMessage;
+  console.log("email, password 검사");
+
+  if (element.value == "") {
+    const message = element === email ? "이메일을 입력해 주세요." : "비밀번호를 입력해 주세요.";
+    toggleErrorMessage(errorMessage, message);
+  } else if (element === email && !element.validity.valid) {
+    toggleErrorMessage(errorMessage, "올바른 이메일 주소가 아닙니다.");
+  } else {
+    toggleErrorMessage(errorMessage);
+  }
+}
+
+//로그인 유효성 검사
+function checkLogin() {
+  const validEmail = "test@codeit.com";
+  const validPassword = "codeit101";
+
+  if (email.value === validEmail && password.value === validPassword) {
+    console.log("로그인 성공");
+    toggleErrorMessage(emailErrorMessage);
+    toggleErrorMessage(passwordErrorMessage);
+    //페이지 이동
+    location.href = "../folder/index.html";
+  } else {
+    console.log("로그인 실패");
+    toggleErrorMessage(emailErrorMessage, "이메일을 확인해 주세요.");
+    toggleErrorMessage(passwordErrorMessage, "비밀번호를 확인해 주세요.");
+  }
+}
+
+//icon-eye 클릭하면 비밀번호 토글
+function togglePassword() {
+  const icon = eyeBtn.firstElementChild;
+
+  console.log("비밀번호 토글");
+  console.log(password.type);
+
+  if (password.type === "password") {
+    password.type = "text";
+    icon.src = "../public/icon/eye-on.svg";
+  } else {
+    password.type = "password";
+    icon.src = "../public/icon/eye-off.svg";
+  }
+}
+
+//사용자 입력 함수 호출 제어
+function handleUserAction(event) {
+  console.log(event.key);
+  if (event.type === "click") {
+    checkLogin();
+  } else if (event.type === "keypress" && event.key === "Enter") {
+    checkLogin();
+    element.preventDefault();
+  }
+}
+
+/********************
+ * EVENT HANDLER
+ ********************/
+
+//focusout 이벤트 핸들러
+email.addEventListener("focusout", function () {
+  checkForm(email);
+});
+
+password.addEventListener("focusout", function () {
+  checkForm(password);
+});
+
+//click, keypress 이벤트 핸들러
+loginBtn.addEventListener("click", handleUserAction);
+document.addEventListener("keypress", handleUserAction);
+eyeBtn.addEventListener("click", togglePassword);
```

Comment:
만약 조금 더 나가서 제가 구조를 짠다면 이런식으로 짜볼것 같아요! 

```js
/**
 * HTML DOM Elements
 */
const emailInput = document.querySelector('#email-input')
const passwordInput = document.querySelector('#password-input')
const loginButton = document.querySelector('#login-button')
const showPasswordButton = document.querySelector(".eyeoff");
/**
 * UI Event Listeners
 */

/**
 * Utility Logics
 */
function isEmailFormat(emailString) {
  const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
  return regex.test(emailString)
}

function isEmpty(text) {
  return text.length === 0
}

/**
 * Validates a user's login information.
 *
 * This function checks if the provided login information matches a predefined valid user schema.
 * It compares both email and password from the loginInfo object with the valid user credentials.
 *
 * @param {Object} loginInfo - An object containing the user's login information.
 * @param {string} loginInfo.email - The email address of the user.
 * @param {string} loginInfo.password - The password of the user.
 * @returns {boolean} - Returns `true` if the email and password match the valid user schema, otherwise returns `false`.
 */
function isValidUser(loginInfo) {
  const validUserSchema = { email: "codeit@codeit.com", password: "password" }

  const { email, password } = loginInfo
  
  if (email !== validUserSchema.email) {
    return false
  }

  if (password !== validUserSchema.password) {
    return false
  }

  return true
}

function validateUserLoginInfo() {
  const email = emailInput.value; 

  const isEmailEmpty = isEmpty(email)
  if (isEmailEmpty) {
    toggleError(emailErrorMessage, "이메일을 입력하세요")
    return false
  }

  const isValidEmailFormat = isEmailFormat(email)
  if (!isValidEmailFormat) {
    toggleError(emailErrorMessage, '잘못된 이메일 유형입니다')
    return false
  }

  const password = passwordInput.value;
  const isPasswordEmpty = isEmpty(password)
  if (isPasswordEmpty) {
    toggleError(passwordErrorMessage, "비밀번호를 입력하세요")
    return false
  }

  return true
}
/**
 * Service Logic
 */
function handleLogin(e) {
  const isValidInput = validateUserLoginInfo()
  if (!isValidInput) {
    return false
  }

  // login 로직 실행
  const user = isValidUser({ email, password });
  
  if (!user) {
    toggleError(emailErrorMessage, "이메일을 확인하세요")
    toggleError(passwordErrorMessage, "비밀번호를 확인하세요")
    return
  }

  location.href = "로그인 성공 후 이동 될 엔드포인트"
}
```